### PR TITLE
[W4.4b][F11-A2]Zhang Yijiong

### PR DESCRIPTION
**Pull request by Zhang-Y-J**
Update address class by adding four more classes inside

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,12 +8,16 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, some street, some unit, some postalcode";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can only be in the format a/BLOCK, STREET, UNIT, POSTAL_CODE";
```
Comment (karrui) [inline src/seedu/addressbook/data/person/Address.java:12]: Avoid lines longer than 110 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -56,4 +74,65 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+
+}
+
+class Block{
```
Comment (karrui) [inline src/seedu/addressbook/data/person/Address.java:80]: These new classes should be its own Java class instead of being a nested class.
Additionally, there are missing header comments for the class. All non trivial classes/ methods should have javadoc format header comments explaining what the class is doing.

File: test/java/seedu/addressbook/commands/DeleteCommandTest.java [hunk 3]
```diff
@@ -33,13 +33,13 @@
     @Before
     public void setUp() throws Exception {
         Person johnDoe = new Person(new Name("John Doe"), new Phone("61234567", false),
-                new Email("john@doe.com", false), new Address("395C Ben Road", false), new UniqueTagList());
+                new Email("john@doe.com", false), new Address("395C, Ben Road, #01-01, 132324", false), new UniqueTagList());
```
Comment (karrui) [inline test/java/seedu/addressbook/commands/DeleteCommandTest.java:36]: [Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! 
If the line has to be split into multiple lines, try to align the parameters.

E.g.
``` 
Person johnDoe = new Person(new Name("John Doe"), 
                 new Phone("61234567", false),
                 ...
```

File: test/java/seedu/addressbook/commands/ViewCommandTest.java [hunk 4]
```diff
@@ -49,7 +49,7 @@ public void execute_personNotInAddressBook_returnsPersonNotInAddressBookMessage(
         ReadOnlyPerson stranger = new Person(new Name("me"),
                                              new Phone("123", true),
                                              new Email("some@hey.go", true),
-                                             new Address("nus", false),
+                                             new Address("0, nus, #01-01, 119224", false),
```
Comment (karrui) [inline test/java/seedu/addressbook/commands/ViewCommandTest.java:52]: Multiple lines should be split like this! :)

File: text1.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+test test
+yj branch
```
Comment (karrui) [inline text1.txt:2]: Avoid committing unrelated files.

**Review by karrui — COMMENTED** [review]
@ZhangYijiong Some comments added. Please close the PR after reading comments.

**Comment by Zhang-Y-J** [comment]
Comments read.
